bundles/org.openhab.binding.airvisualnode/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -90,6 +97,20 @@
 		<state readOnly="true" pattern="%.1f %unit%"/>
 	</channel-type>
 
+	<channel-type id="Pm_10">
+		<item-type>Number:Density</item-type>
+		<label>PM10</label>
+		<description>PM10 level, µg/m&#179;(Only in Pro version)</description>
+		<state readOnly="true" pattern="%.1f %unit%"/>
+	</channel-type>
+
+	<channel-type id="Pm_01">
+		<item-type>Number:Density</item-type>
+		<label>PM0.1</label>
+		<description>PM0.1 level, µg/m&#179;(Only in Pro version)</description>
+		<state readOnly="true" pattern="%.1f %unit%"/>
+	</channel-type>
```

Comment:
If such channels are only available in certain versions of the device, then you should either make a new thing type or remove these channels during handler initialization for devices that don't support it.

But another question I have is whether "Pro version" refers to the device model or the protocol used to communicate with the device. If it is only a protocol then why not enable it by default instead of making it configurable? Is there no way to identify which devices support this protocol automatically?

Reply:
I can't recognize type before providing password and adding thing and can't remove unused channels:
```
thing.getChannels().removeIf(channel -> channel.getLabel().equals("PM0.1") || channel.getLabel().equals("PM10"));

java.lang.UnsupportedOperationException: null
	at java.util.Collections$UnmodifiableCollection.removeIf(Collections.java:1089) ~[?:?]
```
But I remove from config "isProVersion" and recognizing it by protocol: if "measurements" field in Json is array - then it will be Pro version.

Reply:
You have to modify channels by calling `editThing`, do your changes on the thing builder and then call `updateThing`.

Reply:
done